narwhals/utils.py

```diff
@@ -1067,3 +1072,37 @@ def check_column_exists(columns: list[str], subset: list[str] | None) -> None:
     if subset is not None and (missing := set(subset).difference(columns)):
         msg = f"Column(s) {sorted(missing)} not found in {columns}"
         raise ColumnNotFoundError(msg)
+
+
+def _parse_datetime_selector_to_datetimes(
+    time_unit: TimeUnit | Collection[TimeUnit] | None,
+    time_zone: str | timezone | Collection[str | timezone | None] | None,
+    version: Version,
+) -> AbstractSet[Datetime]:
+    # Adapted from polars: https://github.com/pola-rs/polars/blob/725c96009e4c6cb6b05db7f7e33daf3330a4fa35/py-polars/polars/selectors.py#L1340-L1493
+    time_units: list[TimeUnit]
+    if time_unit is None:
+        time_units = ["ms", "us", "ns"]
+    else:
+        time_units = [time_unit] if isinstance(time_unit, str) else list(time_unit)
+
+    time_zones: list[str | timezone | None]
+    if time_zone is None:
+        time_zones = [None]
+    else:
+        time_zones = (
+            [time_zone] if isinstance(time_zone, (str, timezone)) else list(time_zone)
+        )
+
+    if "*" in time_zones:
+        import zoneinfo
+
+        time_zones.extend(list(zoneinfo.available_timezones()))
+        time_zones.remove("*")
```

Comment:
Since we don't allow "*" to be passed to the Datetime constructor as a time zone, I am explicitly creating all the timezones.